### PR TITLE
Skończone zadanie z Adapterem

### PR DESCRIPTION
**Pull request by snapeeek**
Maciej Woźniak 215925

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java [hunk 1]
```diff
@@ -5,13 +5,17 @@
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+import javax.swing.*;
+
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java]: Direct coupling with DrawerFeature class is unnecessary.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java [hunk 2]
```diff
@@ -5,13 +5,17 @@
...
+
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class DrawingControllerAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java]: The name is better but it does not indicate the purpose of the adapter.

Inheritance is not used (no super drawLine calls). Should be removed!

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/SpecialLineAdapter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+import javax.swing.*;
+
+import static edu.kis.powp.jobs2d.features.DrawerFeature.getDrawerController;
+
+/**
+ * driver adapter to drawer with several bugs.
+ */
+public class SpecialLineAdapter extends DrawPanelController implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/SpecialLineAdapter.java]: Similar issues

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -9,13 +9,21 @@
...
+	{
 		this.driverManager = driverManager;
+		this.name = n;
 	}
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+
+		if (name.equals("Joe1"))
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java]: Enum or separate listeners would be better. At least you should consider constants.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java (hunk 2 above)
Comment (snapeeek) [reply src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawingControllerAdapter.java]: Inheritance is used once - in constructor.

**Pull request closed without merging**